File: backend/app/schemas/oauth.py
"""
OAuth2配置相关Schema
"""
from typing import Optional, Dict, Literal
from pydantic import BaseModel, Field


class OAuthConfigUpdate(BaseModel):
    """更新OAuth2配置请求"""
    configType: Literal["system", "chat"] = Field(default="system", description="配置类型: system/chat")
    authorizationUrl: str = Field(..., description="授权端地址")
    tokenUrl: str = Field(..., description="Token端地址")
    userInfoUrl: str = Field(..., description="用户信息端地址")
    scope: str = Field(..., description="连接范围")
    clientId: str = Field(..., description="客户端ID")
    clientSecret: str = Field(..., description="客户端密钥")
    fieldMapping: Optional[Dict] = Field(None, description="字段映射")
    redirectUrl: str = Field(..., description="回调地址")
    enabled: bool = Field(default=False, description="是否启用")


class OAuthConfigResponse(BaseModel):
    """OAuth2配置响应"""
    id: int
    configType: str = "system"
    authorizationUrl: str
    tokenUrl: str
    userInfoUrl: str
    scope: str
    clientId: str
    clientSecret: str
    fieldMapping: Optional[Dict] = None
    redirectUrl: str
    enabled: bool
    createdAt: Optional[str] = None
    updatedAt: Optional[str] = None


class OAuthLoginResponse(BaseModel):
    """OAuth2登录响应"""
    token: str
    expiresIn: int
    user: dict

File: frontend/src/api/oauth.ts
import request from './index'

export interface OAuthConfig {
  id: number
  configType: string
  authorizationUrl: string
  tokenUrl: string
  userInfoUrl: string
  scope: string
  clientId: string
  clientSecret: string
  fieldMapping: Record<string, string> | null
  redirectUrl: string
  enabled: boolean
  createdAt: string | null
  updatedAt: string | null
}

export interface OAuthLoginUrl {
  loginUrl: string
  state: string
}

export interface OAuthCallbackResult {
  token: string
  expiresIn: number
  user: Record<string, any>
}

/**
 * 获取系统OAuth2配置
 */
export function getOAuthConfig(configType: string = 'system') {
  return request.get<never, { code: number; message: string; data: OAuthConfig }>(
    `/oauth/config?config_type=${configType}`
  )
}

/**
 * 保存OAuth2配置
 */
export function saveOAuthConfig(data: Partial<OAuthConfig>) {
  return request.post<never, { code: number; message: string; data: OAuthConfig }>(
    '/oauth/config',
    data
  )
}

/**
 * 获取系统OAuth2授权URL（平台登录用）
 */
export function getOAuthLoginUrl() {
  return request.get<never, { code: number; message: string; data: OAuthLoginUrl }>(
    '/oauth/login-url'
  )
}

/**
 * 系统OAuth2回调处理（平台登录用）
 */
export function handleOAuthCallback(code: string, state?: string) {
  const data: Record<string, string> = { code }
  if (state) {
    data.state = state
  }
  return request.post<never, { code: number; message: string; data: OAuthCallbackResult }>(
    '/oauth/callback',
    data
  )
}

/**
 * 获取对话用户OAuth2授权URL（应用集成用）
 */
export function getChatOAuthLoginUrl() {
  return request.get<never, { code: number; message: string; data: OAuthLoginUrl }>(
    '/oauth/chat-login-url'
  )
}

/**
 * 对话用户OAuth2回调处理（应用集成用）
 */
export function handleChatOAuthCallback(code: string, state?: string) {
  const data: Record<string, string> = { code }
  if (state) {
    data.state = state
  }
  return request.post<never, { code: number; message: string; data: OAuthCallbackResult }>(
    '/oauth/chat-callback',
    data
  )
}
